src/daemon/command_parser_executor.cpp

```diff
@@ -48,6 +51,50 @@ t_command_parser_executor::t_command_parser_executor(
   : m_executor(ip, port, login, ssl_options, is_rpc, rpc_server)
 {}
 
+// Consumes an argument from the given list, if present, parsing it into `var`.
+// Returns false upon parse failure, true otherwise.
+template <typename T>
+static bool parse_if_present(std::forward_list<std::string> &list, T &var, const char *name)
+{
+  if (list.empty()) return true;
+  if (epee::string_tools::get_xtype_from_string(var, list.front()))
+  {
+    list.pop_front();
+    return true;
+  }
+
+  std::cout << "unexpected " << name << " argument: " << list.front() << std::endl;
+  return false;
+}
+
+bool t_command_parser_executor::print_checkpoints(const std::vector<std::string> &args)
+{
+  uint64_t start_height = cryptonote::COMMAND_RPC_GET_CHECKPOINTS::HEIGHT_SENTINEL_VALUE;
+  uint64_t end_height   = cryptonote::COMMAND_RPC_GET_CHECKPOINTS::HEIGHT_SENTINEL_VALUE;
+
+  std::forward_list<std::string> args_list(args.begin(), args.end());
+  bool print_json = !args_list.empty() && args_list.front() == "+json";
+  if (print_json)
+    args_list.pop_front();
+
+  if (!parse_if_present<decltype(start_height)>(args_list, start_height, "start height"))
```

Comment:
Was the the `<decltype(start_height)>` necessary here for some reason?  T should have been deduced here.

Reply:
Odd. Was complaining before it couldn't deduce T, but doesn't complain anymore.